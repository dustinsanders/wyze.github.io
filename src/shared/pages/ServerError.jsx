// @flow

/**
 * Credit for ghost: http://codepen.io/parulbhatti/pen/KzvLYr
 */

import '../styles/global'
import { GhostElectric } from '../components/svgs'
import type { Style, VNode } from '../../types'
import { centered } from '../styles'
import { css } from 'glamor'
import Button from '../components/Button'
import H2 from '../components/H2'
import H3 from '../components/H3'
import preact from 'preact'

const underline: Style = css({
  width: '5em',
  height: '.12em',
  background: '#8bc34a',
  margin: '1em auto',
})

const ServerError = (): VNode => (
  <div {...centered}>
    <GhostElectric />
    <H2>Internal Server Error</H2>
    <div {...underline} />
    <H3>Sorry! Something went wrong on my side.</H3>
    <Button text="Try Again" />
  </div>
)

export default ServerError
