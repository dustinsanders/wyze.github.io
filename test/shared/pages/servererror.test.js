// @flow

import ServerError from '../../../src/shared/pages/ServerError'
import snapshot from '../../helpers/snapshot'

describe('<ServerError />', () => {
  it('renders', snapshot.bind(null, ServerError))
})
