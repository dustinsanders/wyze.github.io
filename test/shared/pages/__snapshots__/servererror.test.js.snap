// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServerError /> renders 1`] = `
"<div data-css-jgfdvp>
  <GhostElectric></GhostElectric>
  <H2>Internal Server Error</H2>
  <div data-css-1mvycfw></div>
  <H3>Sorry! Something went wrong on my side.</H3>
  <Button text=\\"Try Again\\"></Button>
</div>"
`;
